### PR TITLE
Standardize connection format

### LINKED ISSUE #28: Rethinking the target pattern to match on function definitions
See: https://github.com/liveview-native/elixirconf_chess/blob/main/lib/elixirconf_chess_web/live/index_live.ex#L12

example:

```elixir
def render(%{ platform_id: :swiftui, native: %{ platform_config: %{ user_interface_idiom: "watch" } } } = assigns) do
```

While something this complex may be technically correct, it is not meeting our standard of being something easy to use. We should simply have:

```elixir
 def render(%{ platform_id: :swiftui, target_id: :watchos}  = assigns) do
```

Any reason why this isn't a good idea?

### PR DESCRIPTION
**Pull request by supernintendo**
This PR standardizes the initial connection format across platforms. Platform libraries are now expected to pass a map named `_lvn` which supports the following parameters, automatically applying them to `assigns` on mount.

- `:app_version`<sup>1</sup> - The app version as a string
- `:app_build`<sup>1</sup> - The app build as a string
- `:bundle_id` - The app bundle ID as a string (i.e. `com.myapp.mycompany`)
- `:format` - The format of the request, **replaces `platform_id`** (i.e. `:swiftui`, `:jetpack`, etc.)
- `:os`<sup>1</sup> - The OS name as a string
- `:os_version`<sup>1</sup> - The OS version as a string
- `:target`<sup>1</sup> - The client variant (i.e. `:mac`, `:pad`, `:phone`, `:tv`, `:vision`, `:watch`)

<sub>1. These values were previously nested within the `native.platform_config` assign but are now surfaced as top-level assigns.</sub>

Related PR: https://github.com/liveview-native/liveview-client-swiftui/pull/1153
Closes #28

### CODE REVIEW

File: guides/common-features/render-patterns.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ patterns and when you might use them.
 ## `render/1` function clauses
 
 The most common pattern that is often used throughout this guide is the function clause pattern.
-Each function clause matches on the `:platform_id`, allowing each platform to define its own UI:
+Each function clause matches on the `:layout`, allowing each platform to define its own UI:
```
Comment (bcardarella) [inline guides/common-features/render-patterns.md:9]: I'm curious why `layout` instead of `format` here?

Comment (supernintendo) [reply guides/common-features/render-patterns.md]: @bcardarella Oops, good catch. Fixed!

**Review by bcardarella — APPROVED** [review]

**Comment by bcardarella** [comment]
@BrooklinJazz originally we used `platform` to differentiate between the device platforms such as `swiftui` and `jetpack` but internally Phoenix uses `format` for differentiating between `html`, `json`, `xml` type format respones. We decided to normalize on this as it will allow us to opt into auto layout format rendering when LiveView adds support for it

**Comment by supernintendo** [comment]
Waiting on https://github.com/liveview-native/liveview-client-swiftui/pull/1153 to be approved before merging

**Pull request merged**